cpp/src/barretenberg/proof_system/work_queue/work_queue.cpp

```diff
@@ -111,12 +111,15 @@ void work_queue::put_fft_data(barretenberg::fr* result, const size_t work_item_n
         if (item.work_type == WorkType::SMALL_FFT) {
             if (count == work_item_number) {
                 const size_t n = key->circuit_size;
-                barretenberg::polynomial& wire_fft = key->polynomial_cache.get(item.tag + "_fft", 4 * n + 4);
+                barretenberg::polynomial wire_fft(4 * n + 4);
 
                 for (size_t i = 0; i < n; ++i) {
                     wire_fft[4 * i + item.index] = result[i];
                 }
                 wire_fft[4 * n + item.index] = result[0];
+
+                key->polynomial_store.put(item.tag + "_fft", std::move(wire_fft));
```

Comment:
Do we no longer have this polynomial in the store before the computation? Because if we do, then we'll be using extra memory

Reply:
The memory usage is the same with this change. There used to be an overload of `get`of the form `get(key, size)` that would construct a polynomial of the given `size` in place if it did not exist in the cache. So previously, we were constructing a size `4n+4` polynomial of zeros in the cache then setting its values in place. Now I'm constructing that same polynomial locally, then moving it. In either case, we're only storing the memory once at any given time.